src/aead.rs

```diff
@@ -219,30 +377,204 @@ mod public {
 		#[test]
 		fn test_diff_secret_key_err() {
 			let key = SecretKey::default();
-			let plaintext = "Secret message".as_bytes().to_vec();
+			let plaintext = "Secret message".as_bytes();
 
-			let dst_ciphertext = seal(&key, &plaintext).unwrap();
+			let dst_ciphertext = seal(&key, plaintext).unwrap();
 			let bad_key = SecretKey::default();
 			assert!(open(&bad_key, &dst_ciphertext).is_err());
 		}
 
 		#[test]
 		fn test_secret_length_err() {
 			let key = SecretKey::generate(31).unwrap();
-			let plaintext = "Secret message Secret message Secret message Secret message "
-				.as_bytes()
-				.to_vec();
+			let plaintext =
+				"Secret message Secret message Secret message Secret message ".as_bytes();
+
+			assert!(seal(&key, plaintext).is_err());
+			assert!(open(&key, plaintext).is_err());
+		}
+	}
+
+	mod test_stream_seal_open {
+		use super::*;
+
+		#[test]
+		fn test_auth_enc_encryption_decryption() {
+			let key = SecretKey::default();
+			let (mut sealer, nonce) = StreamSealer::new(&key).unwrap();
+			let mut opener = StreamOpener::new(&key, &nonce).unwrap();
+			let plaintext = "Secret message".as_bytes();
+
+			let dst_ciphertext = sealer.seal_chunk(plaintext, StreamTag::MESSAGE).unwrap();
+			assert!(dst_ciphertext.len() == plaintext.len() + 17);
+			let (dst_plaintext, tag) = opener.open_chunk(&dst_ciphertext).unwrap();
+			assert_eq!(plaintext, &dst_plaintext[..]);
+			assert_eq!(tag, StreamTag::MESSAGE);
+		}
+
+		/* TODO, decision about empty ciphertext
+		#[test]
+		fn test_auth_enc_plaintext_empty_err() {
```

Comment:
IMO we should leave it as-is. Supporting empty ciphertexts, but still require the `Tag`. This way we support use cases where someone may need to pass information without any real encrypted data, such as rekey tags during sessions. I believe this is also what you suggested to begin with.